### PR TITLE
Made code that uses the framebuffer more readable

### PR DESCRIPTION
**Pull request by kwagyeman**
Trying to remove MAIN_FB() stuff from being everywhere. Instead, it's inside of access functions.

### CODE REVIEW

**Comment by rcroeder** [comment]
Please remove me from this thread. I have attempted a dozen times to unsubscribe.

**Comment by iabdalkader** [comment]
@rcroeder We didn't add you, check your notifications settings (the watch/unwatch button in the top)

File: src/omv/framebuffer.h [hunk 1]
```diff
@@ -55,8 +45,8 @@ bool fb_get_streaming_enabled();
 int fb_encode_for_ide_new_size(image_t *img);
 void fb_encode_for_ide(uint8_t *ptr, image_t *img);
 
-// Returns the main frame buffer size, factoring in pixel formats.
-uint32_t fb_buffer_size();
+// Initializes an image_t struct with the frame buffer.
+void fb_initialize_image(image_t *img);
```
Comment (iabdalkader) [inline src/omv/framebuffer.h:49]:
```suggestion
void framebuffer_initialize_image(image_t *img);
```

File: src/omv/framebuffer.h [hunk 2]
```diff
@@ -78,4 +68,14 @@ uint8_t *framebuffer_get_buffer();
...
 void framebuffer_set(int32_t w, int32_t h, int32_t bpp);
 
+// Use these macros to get a pointer to main or JPEG framebuffer.
+#define MAIN_FB()           (fb_framebuffer)
+#define JPEG_FB()           (jpeg_fb_framebuffer)
+
+// Use these macros to get the start/end addresses of the main or JPEG framebuffer.
+#define MAIN_FB_START()     (MAIN_FB()->pixels)
+#define JPEG_FB_START()     (JPEG_FB()->pixels)
+#define MAIN_FB_END()       (MAIN_FB()->pixels + framebuffer_get_size())
```
Comment (iabdalkader) [inline src/omv/framebuffer.h:78]:
```suggestion
```

**Comment by iabdalkader** [comment]
Next step, let's rename fb_framebuffer to framebuffer and jpeg_fb_framebuffer to jpegbuffer....

File: src/omv/framebuffer.c [hunk 3]
```diff
@@ -67,90 +70,93 @@ void fb_encode_for_ide(uint8_t *ptr, image_t *img)
...
 
-                // Unlock the framebuffer mutex
-                mutex_unlock(&JPEG_FB()->lock, MUTEX_TID_OMV);
+                mutex_unlock(&jpeg_fb_framebuffer->lock, MUTEX_TID_OMV);
             }
+
             if (does_not_fit) {
-                image_t out = { .w=MAIN_FB()->w, .h=MAIN_FB()->h, .bpp=MAIN_FB()->bpp, .data=MAIN_FB()->pixels };
-                int new_size = fb_encode_for_ide_new_size(&out);
+                printf("Warning: JPEG too big! Trying framebuffer transfer using fallback method!\n");
```
[the remark this answers is not in the document]
Comment (kwagyeman) [reply src/omv/framebuffer.c:118]: The frame rate will drop to 1 fps or so when this code path starts to run. I added this print statement on purpose.

File: src/omv/framebuffer.c [hunk 4]
```diff
@@ -8,10 +8,13 @@
  *
  * Framebuffer functions.
  */
+#include <stdio.h>
```
[the remark this answers is not in the document]
Comment (kwagyeman) [reply src/omv/framebuffer.c:11]: This is there for a reason.

File: src/omv/framebuffer.h [hunk 5]
```diff
@@ -78,4 +68,14 @@ uint8_t *framebuffer_get_buffer();
 // Set the framebuffer w, h and bpp.
 void framebuffer_set(int32_t w, int32_t h, int32_t bpp);
 
+// Use these macros to get a pointer to main or JPEG framebuffer.
+#define MAIN_FB()           (fb_framebuffer)
+#define JPEG_FB()           (jpeg_fb_framebuffer)
+
+// Use these macros to get the start/end addresses of the main or JPEG framebuffer.
+#define MAIN_FB_START()     (MAIN_FB()->pixels)
```
[the remark this answers is not in the document]
Comment (kwagyeman) [reply src/omv/framebuffer.h:76]: Removed

File: src/omv/framebuffer.c [hunk 6]
```diff
@@ -67,90 +70,93 @@ void fb_encode_for_ide(uint8_t *ptr, image_t *img)
...
 void fb_update_jpeg_buffer()
 {
     static int overflow_count = 0;
 
-    if (MAIN_FB()->streaming_enabled && JPEG_FB()->enabled) {
-        if (MAIN_FB()->bpp > 3) {
+    image_t src;
+    fb_initialize_image(&src);
+
+    if (MAIN_FB()->streaming_enabled && jpeg_fb_framebuffer->enabled) {
```
[the remark this answers is not in the document]
Comment (kwagyeman) [reply src/omv/framebuffer.c:101]: Fixed

File: src/omv/fb_alloc.c [hunk 7]
```diff
@@ -42,7 +42,7 @@ void fb_alloc_init0()
 
 uint32_t fb_avail()
 {
-    uint32_t temp = pointer - ((char *) MAIN_FB_PIXELS()) - sizeof(uint32_t);
+    uint32_t temp = pointer - ((char *) MAIN_FB_END()) - sizeof(uint32_t);
```
[the remark this answers is not in the document]
Comment (kwagyeman) [reply src/omv/fb_alloc.c:45]: Fixed

File: src/omv/fb_alloc.c [hunk 8]
```diff
@@ -52,7 +52,7 @@ void fb_alloc_mark()
     char *new_pointer = pointer - sizeof(uint32_t);
 
     // Check if allocation overwrites the framebuffer pixels
-    if (new_pointer < (char *) MAIN_FB_PIXELS()) {
+    if (new_pointer < (char *) MAIN_FB_END()) {
```
[the remark this answers is not in the document]
Comment (kwagyeman) [reply src/omv/fb_alloc.c:55]: Fixed

File: src/omv/framebuffer.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (kwagyeman) [reply src/omv/framebuffer.c:118]: This is on purpose. Won't fix. As mentioned previously... when this code path executes the FPS will be very low which is the reason for the print statement to warn users of this.

File: src/omv/fb_alloc.c [hunk 9]
```diff
@@ -102,7 +102,7 @@ void *fb_alloc(uint32_t size, int hints)
     char *new_pointer = result - sizeof(uint32_t);
 
     // Check if allocation overwrites the framebuffer pixels
-    if (new_pointer < (char *) MAIN_FB_PIXELS()) {
+    if (new_pointer < (char *) MAIN_FB_END()) {
```
[the remark this answers is not in the document]
Comment (kwagyeman) [reply src/omv/fb_alloc.c:105]: Fixed

File: src/omv/fb_alloc.c [hunk 10]
```diff
@@ -138,7 +138,7 @@ void *fb_alloc0(uint32_t size, int hints)
 
 void *fb_alloc_all(uint32_t *size, int hints)
 {
-    uint32_t temp = pointer - ((char *) MAIN_FB_PIXELS()) - sizeof(uint32_t);
+    uint32_t temp = pointer - ((char *) MAIN_FB_END()) - sizeof(uint32_t);
```
[the remark this answers is not in the document]
Comment (kwagyeman) [reply src/omv/fb_alloc.c:141]: Fixed.

File: src/omv/framebuffer.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (kwagyeman) [reply src/omv/framebuffer.c:11]: Need for printing above.

File: src/omv/framebuffer.c [hunk 11]
```diff
@@ -67,90 +70,93 @@ void fb_encode_for_ide(uint8_t *ptr, image_t *img)
     *ptr++ = 0xFE;
 }
 
-uint32_t fb_buffer_size()
+void fb_initialize_image(image_t *img)
```
[the remark this answers is not in the document]
Comment (kwagyeman) [reply src/omv/framebuffer.c:73]: Fixed

File: src/omv/framebuffer.h (hunk 1 above)
Comment (kwagyeman) [reply src/omv/framebuffer.h:49]: Fixed

File: src/omv/framebuffer.h (hunk 2 above)
Comment (kwagyeman) [reply src/omv/framebuffer.h:78]: Removed.

**Comment by kwagyeman** [comment]
It's all been updated.

**Pull request merged**